src/mapwindow.jl

```diff
@@ -49,16 +50,47 @@ and then `mapwindow(f, img, (m,n))` should filter at the 75th quantile.
 
 See also: [`imfilter`](@ref).
 """
-function mapwindow(f, img, window, border="replicate", imginds=indices(img); callmode=:copy!)
-    _mapwindow_kernel(replace_function(f),
+function mapwindow(f, img, window, border="replicate",
+                   imginds=default_imginds(img, window, border); callmode=:copy!)
+    if callmode != :copy!
+        error("Only callmode=:copy! is currently supported")
+    end
+    _mapwindow(f,
               img,
               resolve_window(window),
               resolve_border(border),
-              resolve_imginds(imginds),
-              default_shape(f);
```

Comment:
`default_shape` is used as needed to reshape `buf` before passing it to `f`. One could require that `f` handle whatever shape you pass to it: if the innate implementation of `f` requires an `AbstractVector` input, one could define `f(A) = f(vec(A))`.

Unfortunately, (at least at one point) this led to a performance degradation, because this approach requires that you call `vec` for every output point. The original implementation of `mapwindow` used `vec` once to create a view of `buf` and then passed the view to `f`.

If this wasn't clear, then I should have added more comments. Sorry about that.

Reply:
Yes, `default_shape` still exists, it is used in the `allocate_buffer` function, instead of `mapwindow_kernel!` directly.